UnityProject/Assets/Scripts/Items/Weapons/GunElectrical.cs

```diff
@@ -116,10 +133,32 @@ public override void ServerPerformInteraction(InventoryApply interaction)
 			}
 			else if (Validations.HasItemTrait(interaction.UsedObject, CommonTraits.Instance.Wirecutter) && allowPinSwap)
 			{
-				Inventory.ServerDrop(pinSlot);
+				void ProgressFinishAction()
```

Comment:
why is this a clone with Gun.cs code? Since this class inherits from ``Gun`` you could move this logic to a protected method so you can reuse it in their children.